JAICore/jaicore-algorithminspector/src/main/java/ai/libs/jaicore/graphvisualizer/events/gui/DefaultGUIEventBus.java

```diff
@@ -21,7 +23,11 @@ private DefaultGUIEventBus() {
 
 	@Override
 	public void registerListener(GUIEventListener graphEventListener) {
-		guiEventListeners.add(graphEventListener);
+		if (graphEventListener instanceof IGUIMainPluginController) {
+			guiEventListeners.add(0, graphEventListener);
+		} else {
+			guiEventListeners.add(graphEventListener);
```

Comment:
I am a bit puzzled by this interface. Two points:
1. What is the purpose of this interface? I cannot image why the interface should indicate the role of being the main plugin.
2. and why should a *listener* have the type of a *controller*?

Reply:
All controllers are GraphEventListeners. I chose the interface here as a method declare the main plugin as this one has to be first in the list of listeners. Otherwise, resetting and going to a timestep dont work properly